onnxruntime_extensions/tools/pre_post_processing/steps/nlp.py

```diff
@@ -0,0 +1,508 @@
+
+        input_type_str0, input_shape_str0 = self._get_input_type_and_shape_strs(graph, 0)
+        output_shape_str = "batch_size,max_seq_len"
+
+        def build_output_declare():
+            output_base = []
+            for out in self.output_names:
+                output_base.append(f"{input_type_str0}[{input_shape_str0}] {out}")
+
+            return ",".join(output_base)
+
+        def build_output_imp():
+            output_str = []
+            for idx, out in enumerate(self.output_names):
+                output_str.append(f"{out} = Identity({self.input_names[idx % len(self.input_names)]})")
+
+            output_str = "\n".join(output_str)
+            return output_str
+
+        def build_input_declare():
+            inputs = []
+
+            for idx, inp in enumerate(self.input_names):
+                input_type_str_x, input_shape_str_x = self._get_input_type_and_shape_strs(graph, idx)
+                inputs.append(f"{input_type_str_x}[{input_shape_str_x}] {inp}")
+
+            return ",".join(inputs)
+
+        def build_tokenizer_call_arg():
+            call_args = f"{self.input_names[0]}"
+            return call_args
+
+        converter_graph = onnx.parser.parse_graph(
+            f"""\
+            qa_task ({build_input_declare()}) 
+                => ({build_output_declare()})
+            {{
+                {build_output_imp()}
+            }}
+            """
+        )
+
+        return converter_graph
+
+
+class BertTokenizerQATaskDecoder(Step):
+    def __init__(self, tokenizer_param: TokenizerParam, name: Optional[str] = None):
+        """
+        Brief:
+            Decode the input_ids to text
+        Args:
+            tokenizer_map: some essential infos, If you export tokenizer from hugging-face,
+            such as "tokenizer = transformers.AutoTokenizer.from_pretrained(model_name)"
+            You may need to provide the following:
+                tokenizer_param = TokenizerParam(vocab=tokenizer.vocab, #vocab is dict or file_path)
+            name: Optional name of step. Defaults to 'BertTokenizerQATaskDecoder'
+
+        """
+        # Bugs for default joins in onnxruntime_extensions/tools/pre_post_processing/pre_post_processor.py:187
+        # we have to put "input_ids_1" in the last to avoid failure connected
+        super().__init__(["start_logits", "end_logits", "input_ids_1"], ["text"], name)
+        self._domain = "com.microsoft.extensions"
+        self._tokenizer_param = tokenizer_param
+
+    def _create_graph_for_step(self, graph: onnx.GraphProto, onnx_opset: int):
+        graph_input_names = [inp.name for inp in graph.output]
+
+        # need to reorder input_names by similar to  "input_ids_1", "start_logits", "end_logits"
+        o_inputs = graph_input_names.copy()
+        for inp in o_inputs:
+            if "input_ids_1" in inp:
+                self.input_names[0] = inp
+            elif "start_logits" in inp:
+                self.input_names[1] = inp
```

Comment:
When steps are connected the input names are adjusted to match the predecessor graph, which also involves taking into account any IoMapEntry info. As such I don't think you can expect hard coded names to match, or for the order to match the order of graph.output (e.g. the previous graph may have 6 outputs and we're only using a subset for this step).